### ISSUE TITLE
Add catalog.data.gov host name to catalog-next 

### ISSUE THREAD

**Issue by adborden** [GSA/datagov-ckan-multi#452]
## User Story

As a data.gov operator, I want catalog-next to accept the catalog.data.gov host name so that no application changes are necessary to start serving traffic for catalog.data.gov.

## Acceptance Criteria
- [x] WHEN I request catalog.data.gov from the catalog-web-next host `curl -s -H 'Host: catalog.data.gov' http://localhost/dataset`
  THEN I get a 200
  AND the response contains `data-site-root="https://catalog.data.gov/"`
- [x] PR is ready to merge (but not merged)

## Details / tasks
- [x] Prepare a hotfix PR with an update catalog-next's production.ini and set site_url to catalog.data.gov

We shouldn't actually merge this PR until right before Launch because it might break catalog-next.data.gov. Let's keep the PR in approved draft status, ready to be merged, and update the [Launch Plan](https://github.com/GSA/datagov-ckan-multi/issues/400) to include merging the PR.

**Comment by kimwdavidson** [comment]
Blocked by having a prod environment

**Comment by adborden** [comment]
@thejuliekramer and I confirmed in testing https://github.com/GSA/datagov-ckan-multi/issues/401 that we'll want to update the site url in production.ini right before launch.

**Comment by adborden** [comment]
I think this is already done. https://github.com/GSA/datagov-deploy/pull/2507 is waiting in a draft state until we are ready for launch.
